### ISSUE TITLE
sql: incorrect column name for SHOW STATISTICS

### ISSUE THREAD

**Issue by RaduBerinde** [cockroachdb/cockroach#31923]
The first column returned by SHOW STATISTICS is named incorrectly - this is the statistic name, not the table name.
 
```
root@127.10.190.144:43523/defaultdb> show statistics for table t;
  table_name | column_names |             created              | row_count | distinct_count | null_count | histogram_id  
+------------+--------------+----------------------------------+-----------+----------------+------------+--------------+
  statname   | {"x"}        | 2018-10-26 15:36:31.769624+00:00 |         0 |              0 |          0 |         NULL  
```

**Comment by RaduBerinde** [comment]
Candidate names for the column are `name` (what it used to be), `statistic_name`, `statistics_name`. CC @knz

**Comment by knz** [comment]
woopsie doopsie

statistics_name sounds good

**Comment by RaduBerinde** [comment]
Thanks! It was my bad, I reviewed that change and missed it.

**Comment by RaduBerinde** [comment]
This is being fixed on master by #31927. Leaving the issue open until the backport is in.

**Comment by RaduBerinde** [comment]
Backported in #32045.
